### ISSUE TITLE
MVP for the graph view (node-link diagrams)

### ISSUE THREAD

**Issue by grtlr** [rerun-io/rerun#7897]
This will touch various areas of Rerun so we will split the work up over multiple PRs. This PR is a living document and will be updated to reflect the current status of the implementation.

## MVP for 0.21

Initial PR: #7500 

* [x] Support graphs in the Rerun data model 
* [x] Enable styling of nodes through the Rerun SDK
* [x] Ensure proper bindings for all languages 
* [x] Create Python example for logging graphs
* [x] Remove experimental feature flag
* [x] Add archetypes to `gen_common_index.py`
* [x] Deal with `unused` lint warnings for things that we need in the future
* [x] Fix codegen problem related to `GraphType`
* [x] Add icon for graph view in the UI
* [x] Create `datatype::StringPair` and switch `GraphNode` to `Utf8`
* [x] Update Python example for `Utf8Pair`
* [x] Use `VisualBounds` in Viewer Properties
* [x] Fix binary tree example
  * [x] Implicit nodes are currently on an escape trajectory.
* [x] Entity rect box is drawn incorrectly
* [x] Use a single, top-level `Area` (or no area at all?) (#8227)
* [x] #8282
* [x] #8237
* [x] #8126
* [x] devise how interactive the nodes should be for the map (#8227)
   * instance-based highlight?
   * selection?
   * suppress text selectability?
* [x] #8279
* [x] #8238
* [x] consider making the text constant-UI-point (instead of tight to zoom) (Decision: we are not going to do that, because it is horrible UX). To avoid collisions, users will be able to increase distance between nodes via blueprints.
* [x] #8277
* [x] #8278
* Bugs:
    * [x] Prevent Fjadra crash when nodes go out of bounds of quad tree
    * [x] #8327 
    * [x] Remove hardcoded values from Bezier edges
    * [x] #8456
* [x] Promote `check_graph_time_varying.py` to example

## Nice to have but can live without

* [ ] #8455
* [ ] #8281
* [ ] #8280
* [ ] consider temporarily adding a `EdgeText` component to add labels to edges until tagged components arrive (note: this may be a rabbit hole in terms of layout)

## Future work

These item are not required for the 0.21 release

### Data Model

* [ ] #7916
* [ ] Enable styling of edges through the Rerun SDK (blocked by #6889)
* [ ] Enable linking accross entities, potentially using a `<entity_path>#<node_id>` convention. (#8162)
* [x] Support (statically) logging two archetypes to the same entity (#7992).

### Graph Layout & Viewer

We decided to start with implementing force-based layouts, because they support a wide variety of use cases and allow dynamic/temporal graphs out of the box.

* [x] Support of external layout algorithms through the `positions: Position2D` component (closed by #7500)
* [x] Investigate other layout options in the Rust ecosystem (`fdg`, `layout-rs`, `graphviz-rust`). All of them have some shortcoming for our use cases.
* [x] Re-implement force-based layout from [`d3-force`](https://d3js.org/d3-force). _(⌛ Ongoing)_
  * Implementation currently lives here: https://github.com/grtlr/fjadra, but will likely change to Rerun org. 
  * [x] Re-implement `d3-quadtree` to enable Barnes-Hut approximation 🏎️ 
  * [x] Implement `PositionX` and `PositionY` force
  * [x] Implement `Collide` force
  * [x] Implement `ManyBody` force
  * [x] Implement remaining forces 🦉

### Examples & SDK

* [x] Create C++ example for logging graphs
* [ ] Use `ProjectDirs` to pull in datasets for examples (`social` graph; Observable)
* [ ] #8159

**Comment by abey79** [comment]
Closing this as MVP is complete.
